Source/Services/MethodCaller.cs

```diff
@@ -59,6 +59,27 @@ public MethodCaller(string host, ushort port)
         }
     }
 
+    /// <inheritdoc />
+    public IServerStreamingMethodHandler<TServerMessage> Call<TClientMessage, TServerMessage>(ICanCallAServerStreamingMethod<TClientMessage, TServerMessage> method, TClientMessage request, CancellationToken token)
+        where TClientMessage : IMessage where TServerMessage : IMessage
+    {
+        try
+        {
+            var originalStream = method.Call(request, CreateChannel(), CreateCallOptions(token));
+            return new ServerStreamingMethodHandler<TServerMessage>(
+                new AsyncServerStreamingCall<TServerMessage>(
+                    new CatchingAsyncStreamReader<TServerMessage>(_host, _port, originalStream.ResponseStream),
+                        originalStream.ResponseHeadersAsync,
+                        originalStream.GetStatus,
+                        originalStream.GetTrailers,
+                        originalStream.Dispose));
+        }
+        catch (Exception)
+        {
+            throw new CouldNotConnectToRuntime(_host, _port);
```

Comment:
This is only for one specific exception, we should check only that or just let the original exception through.